internal/fail-printer.go

```diff
@@ -37,6 +37,24 @@ func NewObjectsExpectedActual(expected, actual interface{}) Objects {
 	}
 }
 
+func NewObjectsExpectedActualWithDiff(expected, actual interface{}) Objects {
+	return Objects{
+		NewDiffObject(expected, actual),
```

Comment:
Should we put the diff at the bottom? As the terminal scrolls to the bottom automatically, the user might see it there first, and doesn't have to scroll through the `Actual` and `Expected` sections.